plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt

```diff
@@ -61,17 +63,23 @@ class ToolkitOAuthService : OAuthServiceBase<AccessToken>() {
         return authorize(ToolkitOAuthRequest(registration))
     }
 
-    override fun handleServerCallback(path: String, parameters: Map<String, List<String>>): Boolean {
-        val request = currentRequest.get() ?: return false
-        val toolkitRequest = request.request as? ToolkitOAuthRequest ?: return false
+    override fun handleOAuthServerCallback(path: String, parameters: Map<String, List<String>>): OAuthService.OAuthResult<AccessToken>? {
+        val request = currentRequest.get() ?: return OAuthService.OAuthResult(null, false)
+        val toolkitRequest = request.request as? ToolkitOAuthRequest ?: return OAuthService.OAuthResult(request.request, false)
 
         val callbackState = parameters["state"]?.firstOrNull()
         if (toolkitRequest.csrfToken != callbackState) {
             request.result.completeExceptionally(RuntimeException("Invalid CSRF token"))
-            return false
+            return OAuthService.OAuthResult(toolkitRequest, false)
+        }
+
+        if (parameters["code"] == null) {
+            val error = parameters["error"]?.firstOrNull()
+            val error_description = parameters["error_description"]?.firstOrNull()
+            toolkitRequest.error = OAuthError(error = error, errorDescription = error_description)
```

Comment:
this captures the error callback from the server, but not if the request future was completed exceptionally for any other reason